core/trino-main/src/main/java/io/trino/sql/analyzer/FeaturesConfig.java

```diff
@@ -168,6 +170,40 @@ public boolean canReplicate()
         /**/;
     }
 
+    public enum JoinPushdownMode
+    {
+        OFF,
+        ON,
+        // TODO Add cost based logic to join pushdown
+        // AUTOMATIC,
+        /**/;
+
+        public static JoinPushdownMode fromString(String value)
+        {
+            switch (requireNonNull(value).toLowerCase(Locale.ENGLISH)) {
+                case "false":
+                    return OFF;
+                case "true":
+                    return ON;
```

Comment:
I do not see a need to pretend it's a true/false property.
Why not make it a regular enum property, and then remove those `fromString`, `toString` conversions?

Reply:
changed